NuRadioMC/simulation/simulation.py

```diff
@@ -401,752 +1339,325 @@ def __init__(self, inputfilename,
-                        x2 = self._det.get_relative_position(self._station_id, channel_id) + self._det.get_absolute_position(self._station_id)
-                        logger.debug(f"simulating channel {channel_id} at {x2}")
-
-                        if self._cfg['speedup']['distance_cut']:
-                            t_tmp = time.time()
-                            distance_cut = self._get_distance_cut(shower_energy_sum)
-                            distance = np.linalg.norm(x1 - x2)
-
-                            if distance > distance_cut:
-                                logger.debug('A distance speed up cut has been applied')
-                                logger.debug('Shower energy: {:.2e} eV'.format(self._fin['shower_energies'][self._shower_index] / units.eV))
-                                logger.debug('Distance cut: {:.2f} m'.format(distance_cut / units.m))
-                                logger.debug('Distance to vertex: {:.2f} m'.format(distance / units.m))
-                                distance_cut_time += time.time() - t_tmp
-                                continue
-                            distance_cut_time += time.time() - t_tmp
-
-                        self._raytracer.set_start_and_end_point(x1, x2)
-                        self._raytracer.use_optional_function('set_shower_axis', self._shower_axis)
-                        if pre_simulated and ray_tracing_performed and not self._cfg['speedup']['redo_raytracing']:  # check if raytracing was already performed
-                            if self._cfg['propagation']['module'] == 'radiopropa':
-                                logger.error('Presimulation can not be used with the radiopropa ray tracer module')
-                                raise Exception('Presimulation can not be used with the radiopropa ray tracer module')
-                            sg_pre = self._fin_stations["station_{:d}".format(self._station_id)]
-                            ray_tracing_solution = {}
-                            for output_parameter in self._raytracer.get_output_parameters():
-                                ray_tracing_solution[output_parameter['name']] = sg_pre[output_parameter['name']][self._shower_index, channel_id]
-                            self._raytracer.set_solution(ray_tracing_solution)
+                # group events into events based on signal arrival times
+                events = group_into_events(station, event_group, particle_mode, self._config['split_event_time_diff'], time_logger=self.__time_logger)
+
+                evt_group_triggered = False
+                for evt in events:
+                    station = evt.get_station()
+                    apply_det_response(evt, self._det, self._config, self.detector_simulation_filter_amp,
+                                       bool(self._config['noise']),
+                                       self._Vrms_efield_per_channel, self._integrated_channel_response,
```

Comment:
I am not sure what is correct but in the previous version we used here `self._Vrms_per_channel` instead of `self._Vrms_efield_per_channel`?

Reply:
Thanks for noticing that. I think this was actually a bug but wasn't caught by any unit test because we never have any amplification and therefore the two variables were the same.